pkg/controllers/vdb/onlineupgrade_reconciler.go

```diff
@@ -941,20 +1010,50 @@ func (r *OnlineUpgradeReconciler) redirectConnectionsToReplicaGroupB(ctx context
 		return ctrl.Result{}, err
 	}
 
-	// In lieu of the redirect, we are simply going to update the service object
-	// to map to nodes in replica group B. There is no state to check to avoid
-	// this function. The updates themselves are idempotent and will simply be
-	// no-op if already done.
-	// Routing is easy since the sandbox is a copy of the main cluster.
-	//
+	// remove client routing labels from replica group a
+	actor := MakeClientRoutingLabelReconciler(r.VRec, r.Log, r.VDB, r.PFacts[vapi.MainCluster], DrainNodeApplyMethod, "")
```

Comment:
Why have you done this here?

Reply:
we want connections to be routed to the old cluster until the new one is ready to accept connections. the old main will pause connections it receives and redirect them by the end of this method. you're correct that we should add the routing labels to the new cluster first so that there's not a gap between the old cluster accepting and pausing connections and the new cluster accepting connections